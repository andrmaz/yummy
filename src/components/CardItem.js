import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { Box, Image, Badge, List, ListItem, useColorMode } from '@chakra-ui/react';

// Check on props change
export function areEqual(prevProps, nextProps){
    if (prevProps === nextProps) {
        return true;
    } else {
        return false;
    }
};

const CardItem = ({ recipe }) => {

    // To manage color mode chakra exposes the useColorMode
    const { colorMode } = useColorMode();

    return (
        <Box borderWidth="1px" borderRadius="lg" overflow="hidden" bgColor={colorMode === 'light' ? 'royalblue' : 'unset'}>
            <Image src={recipe.image} alt='Recipe' m='auto' />

            <Box p="6">
                <Box d="flex" alignItems="baseline">
                <Badge borderRadius="full" px="2" colorScheme="teal">
                    {recipe.dietLabels[0]}
                </Badge>
                <Box
                    color="gray.500"
                    fontWeight="semibold"
                    letterSpacing="wide"
                    fontSize="xs"
                    textTransform="uppercase"
                    ml="2"
                >
                    {recipe.totalTime} minutes &bull; {recipe.calories.toFixed(0)} cal
                </Box>
                </Box>

                <Box
                mt="1"
                fontWeight="semibold"
                as="h4"
                lineHeight="tight"
                isTruncated
                >
                {recipe.label}
                </Box>

                <List>
                    {
                        recipe.ingredientLines.map((ingredient, index) => (
                            <ListItem key={index}>
                                {ingredient}
                            </ListItem>
                        ))
                    }
                </List>
            </Box>
        </Box>
    );
};

CardItem.propTypes = {
    recipe: PropTypes.object
};

export default memo(CardItem, areEqual);