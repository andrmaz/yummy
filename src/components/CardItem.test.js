import React from 'react';
import { shallow } from 'enzyme';
import CardItem, { areEqual } from './CardItem';

let wrapper;
const mockRecipe = {
    image: '/',
    dietLabels: ['label'],
    totoalTime: 1,
    calories: 1,
    label: 'label',
    ingredientLines: ['ingredient']
};

describe('CardItem Component', () => {
    beforeEach(() => {
        wrapper = shallow(<CardItem recipe={mockRecipe} />);
        expect.hasAssertions();
    });
    it('should render Component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('should not have set bgColor on dark theme', () => {
        expect(wrapper.prop('bgColor')).toEqual('unset');
    });
    it('should return true if props do not change', () => {
        expect(areEqual(1, 1)).toBeTruthy();
    });
    it('should return false on props change', () => {
        expect(areEqual(1, 0)).toBeFalsy();
    });
});

