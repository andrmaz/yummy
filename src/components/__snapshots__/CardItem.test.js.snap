// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardItem Component should render Component correctly 1`] = `
<Box
  bgColor="unset"
  borderRadius="lg"
  borderWidth="1px"
  overflow="hidden"
>
  <Image
    alt="Recipe"
    m="auto"
    src="/"
  />
  <Box
    p="6"
  >
    <Box
      alignItems="baseline"
      d="flex"
    >
      <Badge
        borderRadius="full"
        colorScheme="teal"
        px="2"
      >
        label
      </Badge>
      <Box
        color="gray.500"
        fontSize="xs"
        fontWeight="semibold"
        letterSpacing="wide"
        ml="2"
        textTransform="uppercase"
      >
         minutes â€¢ 
        1
         cal
      </Box>
    </Box>
    <Box
      as="h4"
      fontWeight="semibold"
      isTruncated={true}
      lineHeight="tight"
      mt="1"
    >
      label
    </Box>
    <List>
      <ListItem
        key="0"
      >
        ingredient
      </ListItem>
    </List>
  </Box>
</Box>
`;
